ouroboros-network-framework/src/Ouroboros/Network/InboundGovernor.hs

```diff
@@ -112,9 +112,10 @@ inboundGovernor trTracer tracer serverControlChannel inboundIdleTimeout
        (\(e :: SomeAsyncException) -> do
          state <- atomically $ readTVar st
          _ <- Map.traverseWithKey
-               (\connId _ ->
+               (\connId _ -> do
+                 let state' = unregisterConnection connId state
```

Comment:
Oh this is just a better way to trace the `RemoteTransitionTrace` without the `emptyState` hack. Basically we want to remove the connection from the state and then `mkRemoteTransitionTrace` will look into the difference and trace the correct transition

Reply:
Addressed

Reply:
If we want to do that properly, we should thread the state and use `Map.foldrWithKey` instead, otherwise ignoring `state'` is just confusing.

Reply:
Just noticed replies by email do not go to the correct thread :/

Reply:
> If we want to do that properly, we should thread the state and use Map.foldrWithKey instead, otherwise ignoring state' is just confusing.

> We are not ignoring state' we are using it to build the trace. That's all
that matters here because the inbound governor is going to die. What we had
previously was a hack and I was building the trace using the empty state,
with this patch it is a cleaner way to do it.

All we need is to remove that particular connection, I understand now what you are saying about threading state, but it is not really necessary, here.

Reply:
right, also every next `unregisterConnection` will look through smaller map (though performance is not critical here as well).